### PR TITLE
Fix test_quantization with Pytorch 1.10

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?

https://github.com/pytorch/pytorch/pull/63043 broke these tests by changing the default of `get_default_qat_qconfig` to use `FusedMovingAvgObsFakeQuantize` rather than `FakeQuantize` . This PR ensures we go back to using `FakeQuantize` as before.

See https://github.com/pytorch/pytorch/issues/64564 for full discussion

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/quantization.py [hunk 1]
```diff
@@ -188,7 +189,11 @@ def on_fit_start(self, trainer, pl_module):
             if self._observer_type == "histogram":
                 pl_module.qconfig = torch.quantization.get_default_qconfig(self._qconfig)
             elif self._observer_type == "average":
-                pl_module.qconfig = torch.quantization.get_default_qat_qconfig(self._qconfig)
+                # Needed because of https://github.com/pytorch/pytorch/issues/64564
+                if _TORCH_GREATER_EQUAL_1_10:
+                    pl_module.qconfig = torch.quantization.get_default_qat_qconfig(self._qconfig, None)
```
Comment (Borda) [inline pytorch_lightning/callbacks/quantization.py:194]: is the None positional argument?
we can also have just something like
```
extra_args = dict(not_sure_name=None) if _TORCH_GREATER_EQUAL_1_10 else {}
pl_module.qconfig = torch.quantization.get_default_qat_qconfig(self._qconfig, **extra_args)
```

**Comment by daniellepintz** [comment]
@Borda it seems there is some issue with the `_TORCH_GREATER_EQUAL_1_10 ` flag?

Asserting it in test_quantization is failing in the PT1.10 CI:
```
def test_quantization(tmpdir, observe: str, fuse: bool, convert: bool):
        """Parity test for quant model."""
        seed_everything(42)
>       assert _TORCH_GREATER_EQUAL_1_10
E       assert False

tests/callbacks/test_quantization.py:38: AssertionError
```

**Comment by Borda** [comment]
> Locally it passes even when using the 1.11 nightly `1.11.0.dev20210930`, so I am a bit confused

can you try locally the `_TORCH_GREATER_EQUAL_1_10` become true for you?

**Comment by daniellepintz** [comment]
yes, it did, since I added `assert _TORCH_GREATER_EQUAL_1_10` to test_quantization

**Comment by daniellepintz** [comment]
```
(base) daniellepintz@ip-10-200-89-146:~/repos/pytorch-lightning$ python
Python 3.9.5 (default, Jun  4 2021, 12:28:51)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.__version__
'1.11.0.dev20210930'
>>>
KeyboardInterrupt
>>>
[2]+  Stopped                 python
(base) daniellepintz@ip-10-200-89-146:~/repos/pytorch-lightning$ python -m pytest -v tests/callbacks/test_quantization.py::test_quantization[False-True-average]
=========================================== test session starts ===========================================
platform linux -- Python 3.9.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /fsx/users/daniellepintz/conda/bin/python
cachedir: .pytest_cache
rootdir: /data/home/daniellepintz/repos/pytorch-lightning, configfile: setup.cfg
plugins: hydra-core-1.1.1
collected 1 item

tests/callbacks/test_quantization.py::test_quantization[False-True-average] PASSED                  [100%]

===================================== 1 passed, 14 warnings in 5.98s ======================================
```

**Review by awaelchli — APPROVED** [review]

File: tests/callbacks/test_quantization.py [hunk 2]
```diff
@@ -34,6 +35,7 @@
 def test_quantization(tmpdir, observe: str, fuse: bool, convert: bool):
     """Parity test for quant model."""
     seed_everything(42)
+    assert _TORCH_GREATER_EQUAL_1_10
```
Comment (carmocca) [inline tests/callbacks/test_quantization.py:38]: Is this here for debugging? This test is not set to run only on 1.10

Comment (carmocca) [reply tests/callbacks/test_quantization.py:38]: Marking this as a draft then.

> do you have any idea why?

Yes. It's a problem with our `_compare_version` implementation and the torch version:

```python
import operator
from packaging.version import Version

# current behaviour in master
x = operator.ge(Version("1.10.0.dev20210922"), Version("1.10.0"))
print(x)  # False

print(Version("1.10.0.dev20210922").base_version)
# 1.10.0

# potential fix
x = operator.ge(Version(Version("1.10.0.dev20210922").base_version), Version("1.10"))
print(x)
# True
```

We should use `base_version` instead

File: pytorch_lightning/callbacks/quantization.py (hunk 1 above)
Comment (ananthsub) [inline pytorch_lightning/callbacks/quantization.py:194]: Could you add a comment for what None corresponds to here as well? It'd save readers one hop of going through the quantization APIs

**Review by ananthsub — APPROVED** [review]

**Comment by daniellepintz** [comment]
All tests green!! Except "verify linking"
